src/core/draw_tools/index.ts

```diff
@@ -65,7 +65,7 @@ class DrawToolsControllerImpl implements DrawToolsController {
       toolboxAtom.setSettings({
         availableModes: ['DrawPolygonMode', 'DrawLineMode', 'DrawPointMode'],
         finishButtonCallback: () => {
-          this.deferred?.resolve(this.geometry);
+          this.deferred?.resolve(this.geometry.features.length ? this.geometry : null);
```

Comment:
we have some helper funcs than you maybe want to refactor/reuse
src/core/bivariate/createBivariateGraphQLQuery.ts#L21
src/utils/geoJSON/helpers.ts#L91

Reply:
Wrote a new version of isGeoJSONEmpty() with unit tests, applied it